mysql-test/main/rowid_filter_innodb.test

```diff
@@ -699,3 +699,61 @@ DROP TABLE t1,t2;
 set global innodb_stats_persistent= @stats.save;
 
 --echo # End of 10.4 tests
+
+--echo #
+--echo # MDEV-31154: Fatal InnoDB error or assertion `!is_v' failure upon multi-update with indexed virtual column
+--echo #
+
+--echo # Test with auto generated Primary Key
+--echo #
+create table t1 (a int, b int as (a/2) virtual, key (b), key (a)) engine=innodb;
+insert into t1(a)
+select (rand(1)*1000)/10 from seq_1_to_1000;
+analyze table t1;
+
+let $q1= select count(*) from t1 where a between 21 and 30 and b=12;
```

Comment:
I don't like the random set either but this is the case I found out from the same test file that traverses the path every time. 
```
1a518aa1d297 (Igor Babaev       2019-06-25 08:06:27 -0700  82) --echo #
1a518aa1d297 (Igor Babaev       2019-06-25 08:06:27 -0700  83) --echo # MDEV-19820: use of rowid filter for innodb table without primary key
1a518aa1d297 (Igor Babaev       2019-06-25 08:06:27 -0700  84) --echo #
1a518aa1d297 (Igor Babaev       2019-06-25 08:06:27 -0700  85)
1a518aa1d297 (Igor Babaev       2019-06-25 08:06:27 -0700  86) create table t1 (a int, b int, key (b), key (a)) engine=innodb;
1a518aa1d297 (Igor Babaev       2019-06-25 08:06:27 -0700  87) insert into t1
58cd0bd59ef0 (Igor Babaev       2022-10-17 16:44:10 -0700  88) select (rand(1)*1000)/10, (rand(1001)*1000)/20 from seq_1_to_1000;
1a518aa1d297 (Igor Babaev       2019-06-25 08:06:27 -0700  89) analyze table t1;
```
The test already uses "set optimizer_switch='rowid_filter=on';" and this certainly is not enough. Can you please elaborate how FORCE can be used to predictably force rowid filter optimization.

Let's say we have 
`create table t1 (a int, b int as (a/2) virtual, key (b), key (a)) engine=innodb;
`
we would like to have the row ID filter plan like
```
id	select_type	table	type	possible_keys	key	key_len	ref	rows	filtered	Extra
1	SIMPLE	t1	ref|filter	b,a	b|a	5|5	const	27 (10%)	9.60	Using where; Using rowid filter
```
set optimizer_switch='rowid_filter=on'
How do we use FORCE to achieve it ?

https://mariadb.com/kb/en/index-hints-how-to-force-query-plans/ shows how index scan can be forced but it don't see how to use for current case.

I could not find any rowId filter mtr test case that using it this way either. Do you have some examples  ?

Reply:
How about this
```sql
--source include/have_innodb.inc
--source include/have_sequence.inc
--source include/innodb_stable_estimates.inc

create table t1 (a int, b int as (a/2) virtual, filler1 char(200), key (b), key (a)) engine=innodb;
insert into t1 (a, filler1) select seq, seq from seq_1_to_10000;
explain select * from t1 where a<1000 and b <1000;

drop table t1;
```
it gives me rowid filter on both 10.6 and 11.0:
```
id      select_type     table   type    possible_keys   key     key_len ref     rows    Extra
1       SIMPLE  t1      range|filter    b,a     a|b     5|5     NULL    999 (20%)       Using index condition; Using where; Using rowid filter
```

Reply:
@spetrunia Thanks for the tests. I could repeat the issue after minor modification to make a/b so that the push down scan is on virtual column index.

I have created a new pull request against 10.4 as it is now repeatable on 10.4.
https://github.com/MariaDB/server/pull/3118